src/shared/materials/weakly_compressible_fluid.cpp

```diff
@@ -40,5 +40,20 @@ Real SymmetricTaitFluid::getSoundSpeed(Real p, Real rho)
                ? sqrt((p0_ + Real(gamma_) * p) / rho)
                : sqrt((p0_ - Real(gamma_) * p) / rho);
 }
+//=================================================================================================//
+Real HerschelBulkleyFluid::getViscosity(Real shear_rate)
+{
+
+    Real effective_shear_rate = SMAX(SMIN(shear_rate, max_shear_rate_), min_shear_rate_);
```

Comment:
```suggestion
    Real effective_shear_rate = std::clamp(shear_rate, min_shear_rate_, max_shear_rate_);
```

Reply:
I will change it in the next pull request.